Tnolet docs update follow up (#655)
checkly/checklyhq.com: site/content/docs/cli/cli-vs-terraform-pulumi.md

---
title: CLI vs. Terraform & Pulumi
weight: 6
menu:
  docs:
    parent: "CLI"
---

Creating, debugging and managing E2E and synthetics monitoring at scale is best done "as code". Currently, Checkly supports

> three tools you can use for your Monitoring-as-Code workflow:

Reviewer: ```suggestion
three tools you can use for your **Monitoring as Code** workflow:
```

Just following the formatting

1. The Checkly CLI.
2. Terraform (through our Terraform provider)
3. Pulumi (through our Pulumi provider)

In most cases, the choice depends on what your goals are and how your organization is set up.

We believe the **Checkly CLI delivers the best-of-breed workflow**. However, there are always trade-offs. Let's list some 
pros and cons based on our own experience and user feedback.

> ## Terraform vs. Checkly CLI

Reviewer: These sections are no real comparisons. Should we just have sections with pros and cons?

## Checkly CLI - Pro / Cons
## TF - Pro / Cons
## Pulumi - Pro / Cons

Author: I like it more like this. I think the are comparisons in the form of pros and cons list.

Reviewer: Fair enough. :) Still think though, that this page isn't explaining when / where our CLI is better and shines. Maybe more intro and CLI explanation, in the beginning, would do the job.

> If you are a Terraform shop, you can use Checkly to its full capacity. The CLI does add core capabilities like TS/JS coding 

Reviewer: ```suggestion
If you are a Terraform shop, you can use Checkly to its full capacity. The CLI adds core capabilities like TS/JS coding 
```

> , simple Git integration and executing tests.

Reviewer: ```suggestion
, simple Git integration and test execution.
```

**Pros**

`+` Terraform integrates very well with Checkly. We have a well maintained Terraform provider.

`+` Terraform is comfortable for many Devops engineers used to writing HCL scripts.

`+` Checkly can be configured almost completely through Terraform resources.

**Cons**

> `-` Terraform does not allow you to test checks or use TS/JS programming.

Reviewer: We probably should rephrase the part above which states `executing tests`.

Reviewer: Hah! I misread this sentence.

> If you are a Terraform shop, you can use Checkly to its full capacity. The CLI adds core capabilities like TS/JS coding 

Maybe we should call it `the Checkly CLI`?
